src/analysis/functions/mod.rs

```diff
@@ -0,0 +1,83 @@
+use r2api::structs::LSectionInfo;
+use petgraph::graph::NodeIndex;
+
+use frontend::radeco_containers::{RadecoFunction, RadecoModule};
+use middle::ir::MOpcode;
+use middle::ssa::ssa_traits::SSA;
+use middle::ssa::ssa_traits::SSAWalk;
+use middle::ssa::ssa_traits::NodeType;
+use middle::regfile::SubRegisterFile;
+
+use std::sync::Arc;
+use std::marker::PhantomData;
+
+pub struct FuncInfo {
+    arguments : Vec<NodeIndex>
+}
+
+pub trait FuncAnalysis: Transfer {}
+
+impl<T: Transfer> FuncAnalysis for T {}
+
+pub trait Transfer {
```

Comment:
Ok, so you don't need to redefine the trait here. The idea behind the `Transfer` and `Propagate` trait defined [here](https://github.com/radare/radeco-lib/blob/fix_panics/src/analysis/reference_marking/reference_marking_inter.rs#L24) is to turn any intra-function analysis, into an inter-function analysis simply by implementing those two traits (without having to implement the machinery to do this for every single analysis that we write). 

The `InterProcAnalysis`, is broken down into two traits: `Transfer` -- How to analyze a single function, and `Propagate` -- What and how to *propagate* this information to other functions.

Here's the idea, `Transfer::transfer` does the first pass of (whatever) analysis, and returns the analyzer instance. `Transfer::transfer_iterative` is then used in all subsequent calls (when the analysis isn't complete and can be improved from the inter-function step). The `Propagate` trait is used to describe how to return the information from the analysis and *propagate* this information to all the callers/callees (at callsites). The actual propagation (along with other low-level handling such as threading, checking for fixpoints etc.) is automatically done by the `InterFunctionAnalyzer`. So which implementing these traits you only have to think about one function at a time.

What you should do is either:

1. Implement the analysis on a struct, and implement the trait `analysis::reference_marking::reference_marking_inter::Transfer` to act like a wrapper and call your analysis functions. For an example, see [reference_marking_intra.rs](https://github.com/radare/radeco-lib/blob/fix_panics/src/analysis/reference_marking/reference_marking_intra.rs). The analysis there can be made inter-function "automatically" by implementing transfer and propagate, where `Transfer::transfer` will simply act as a wrapper and call `resolve_references` (similarly, transfer iterative calls `resolve_references_iterative`).

2. Or, you can skip implementing the analysis on the struct and simply implement the analysis as a part of your trait implementation on the struct, i.e., in the above example, write the code that is a part of resolve_references(/_iterative) directly in transfer(/_iterative) when implementing the transfer trait.

Both these options are equally good. I chose the first one (and encourage you to do the same) as its a bit easier to test out initially, especially when the inter-function analysis is still under development.